sdk/cosmos/azure_data_cosmos/src/location_cache.rs

```diff
@@ -0,0 +1,570 @@
+                    "Location 1".to_string(),
+                    "https://location1.documents.azure.com".to_string()
+                ),
+                (
+                    "Location 2".to_string(),
+                    "https://location2.documents.azure.com".to_string()
+                )
+            ])
+        );
+
+        assert_eq!(
+            cache.locations_info.available_read_endpoints_by_location,
+            HashMap::from([
+                (
+                    "Location 1".to_string(),
+                    "https://location1.documents.azure.com".to_string()
+                ),
+                (
+                    "Location 2".to_string(),
+                    "https://location2.documents.azure.com".to_string()
+                ),
+                (
+                    "Location 3".to_string(),
+                    "https://location3.documents.azure.com".to_string()
+                ),
+                (
+                    "Location 4".to_string(),
+                    "https://location4.documents.azure.com".to_string()
+                )
+            ])
+        );
+
+        assert_eq!(
+            cache.locations_info.write_endpoints,
+            vec![
+                "https://location1.documents.azure.com".to_string(),
+                "https://location2.documents.azure.com".to_string()
+            ]
+        );
+
+        assert_eq!(
+            cache.locations_info.read_endpoints,
+            vec![
+                "https://location1.documents.azure.com".to_string(),
+                "https://location2.documents.azure.com".to_string(),
+            ]
+        );
+    }
+
+    #[test]
+    fn location_cache_update_with_one_preferred_location() {
+        let (default_endpoint, write_locations, read_locations, preferred_locations) =
+            create_test_data();
+
+        let mut cache = LocationCache::new(
+            default_endpoint.to_string(),
+            vec![preferred_locations[0].clone()],
+        );
+
+        cache.update(write_locations, read_locations);
+
+        assert_eq!(
+            cache.default_endpoint,
+            "https://default.documents.azure.com"
+        );
+
+        assert_eq!(
+            cache.locations_info.preferred_locations,
+            vec![preferred_locations[0].clone()]
+        );
+
+        assert_eq!(
+            cache.locations_info.write_endpoints,
+            vec!["https://location1.documents.azure.com".to_string()]
+        );
+
+        assert_eq!(
+            cache.locations_info.read_endpoints,
+            vec!["https://location1.documents.azure.com".to_string()]
+        );
+    }
+
+    #[test]
+    fn mark_read_endpoint_unavailable_test() {
+        // set up test data
+        let (default_endpoint, write_locations, read_locations, preferred_locations) =
+            create_test_data();
+
+        let mut cache = LocationCache::new(default_endpoint, preferred_locations);
+        cache.update(write_locations, read_locations);
```

Comment:
nit: You could probably bundle this code up into a `create_test_location_cache` function to save a bit of repetition later, since you frequently set up a Location Cache with all the test data in it. You'd still keep `create_test_data` separate since you use it on it's own in a few places too.